src/client/app/effects/github.ts

```diff
@@ -109,9 +110,15 @@ export class GitHubEffects {
             }
 
             return this._github.createOrUpdateGist(
-                description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
+                    description, files, gistId, type === GitHub.GitHubActionTypes.SHARE_PUBLIC_GIST
             )
-            .map((gist: IGist) => ({ gist: gist, snippetId: id }));
+            .map((gist: IGist) => ({ gist: gist, snippetId: id }))
+            .catch(exception => {
+                if (exception.status === 404) {
```

Comment:
By the way, if the error doesn't provide you enough information, you may consider doing a pre-validation as a separate call (instead of watching failure on update, to first find the owner ID via an API call to github, and so handle it as normal logic rather than an exception).  I'm fine with either way, as long as we can have enough info.